h/streamer/messages.py

```diff
@@ -91,33 +93,51 @@ def handle_annotation_event(message, socket):
     annotation event, otherwise a dict containing information about the event.
     """
     action = message['action']
-    annotation = storage.annotation_from_dict(message['annotation'])
 
     if action == 'read':
         return None
 
     if message['src_client_id'] == socket.client_id:
         return None
 
-    if annotation.get('nipsa') and (
-            socket.request.authenticated_userid != annotation.get('user', '')):
+    # We don't send anything until we have received a filter from the client
+    if socket.filter is None:
         return None
 
-    if not _authorized_to_read(socket.request, annotation):
+    notification = {
+        'type': 'annotation-notification',
+        'options': {'action': action},
+    }
+    id_ = message['annotation_id']
+
+    # Return early when action is delete
+    if action == 'delete':
+        notification['payload'] = [{'id': id_}]
+        return notification
```

Comment:
Hmm. Does the filter code really require an annotation object, or is a dictionary good enough? If the latter, we could keep sending the serialized annotation only for deletes as a temporary measure. It's ugly as hell, but...